mass_mailing_switzerland/models/mass_mailing_campaign.py

```diff
@@ -26,9 +26,17 @@ class MassMailingCampaign(models.Model):
         'correspondence', related='campaign_id.correspondence_ids'
     )
     invoice_line_ids = fields.One2many(
-        'account.invoice.line', related='campaign_id.invoice_line_ids'
+        'account.invoice.line', compute='_compute_campaign_invoices'
     )
 
+    @api.multi
+    def _compute_campaign_invoices(self):
+        for campaign in self:
+            mass_medium = self.env.ref(
```

Comment:
Sorry I didn't see this but you can move this line outside the for loop to avoid recomputing it at each iteration.